Lib/shutil.py

```diff
@@ -1183,6 +1185,10 @@ def _unpack_zipfile(filename, extract_dir):
                         open(targetpath, 'wb') as target:
                     copyfileobj(source, target)
     finally:
+        if skipped:
+            import warnings
+            warnings.warn(f'unpack {filename}: {skipped} file(s) skipped'
+                          ' (due to absolute path or `..` path component)')
```

Comment:
warnings are a mecanism for a module author to communicate about bad usage to other developers calling the code, not to communicate with end-users: https://docs.python.org/3/howto/logging.html#when-to-use-logging

Reply:
So using logging.warning() here should be appropriate?

Reply:
I looked for a similar usages in the library and found this:
https://github.com/python/cpython/blob/f42a06ba279c916fb67289e47f9bc60dc5dee4ee/Lib/zipfile.py#L1480

it's very similar in that both cases warn about something being lost when creating / unpacking a zip file: some contained files and a part of a comment respectively. That warning was added in 2014.

Reply:
Good find!  @serhiy-storchaka added that change, maybe he can comment here.

Reply:
I just replaced existing prints with warnings. I am not sure that it is a good solution, but it at least gives the user some control: warnings can be printed, silenced or converted to exceptions.

Reply:
I found another example where warning would be for the end user rather than app author: https://github.com/pganssle/zoneinfo/blob/c54d700fcba78b20733390ee3014809ad69858f6/src/backports/zoneinfo/_tzpath.py#L58

-- this one is from 2020 by Paul Ganssle

However I think for this use case, I don't see any issue with using logging.warning or logging.error. But I wonder which one of those would be more appropriate. It fits the description of logging.error more closely but at the same time treating it as an "error" might be too strong.

Reply:
In general `logging.warning` would be cleaner here than using the `warnings` module. As Éric says, this is a data-driven problem that is of interest to the end-user, less so to the developer of the application.

Reply:
@akulakov you could in fact ask on python-dev about the general practice if you want feedback; but I don't think this is particularly controversial: libraries can log things to end-users, and many do. I'd just use the logging module here, and as a follow-up task, fix the other warnings call you found in `zipfile`.

Reply:
@ambv: I agree; I was going to update it but got a little distracted -- will push an update today or tomorrow.. Thanks for looking at this as well!

Reply:
A few functions in shutil.py accept logger as an arg and use it to print info/debug logs. I think in this case there's no reason to follow this pattern because we want the warnings to be displayed by default.